spring-rabbit/src/main/java/org/springframework/amqp/rabbit/RabbitConverterFuture.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.rabbit.listener.DirectReplyToMessageListenerContainer.ChannelHolder;
+import org.springframework.core.ParameterizedTypeReference;
+
+/**
+ * A {@link RabbitFuture} with a return type of the template's
+ * generic parameter.
+ * @param <C> the type.
+ * @since 1.6
+ */
+public class RabbitConverterFuture<C> extends RabbitFuture<C> {
+
+	private volatile ParameterizedTypeReference<C> returnType;
+
+	RabbitConverterFuture(String correlationId, Message requestMessage,
```

Comment:
Looks like those classes in the `AsyncRabbitTemplate` come with a `public` ctor...
Is there a particular reason to change that matter now, since I believe in `3.0` we are going to use this top-level classes instead in a new version of the  `AsyncRabbitTemplate`?
Or was that just a mistake from day first when those inner classes were introduced since IDE just generates ctors with `public` by default?..

Reply:
It was a mistake - they are not static classes so they can't be instantiated outside of the template scope anyway.